tests/npc_test.cpp

```diff
@@ -7,8 +7,23 @@
 #include "map.h"
 #include "text_snippets.h"
 
+#include <limits.h>
 #include <string>
 
+template<typename Val>
+struct range
+{
+    range() :
+        min( std::numeric_limits<Val>::min() ),
+        max( std::numeric_limits<Val>::max() ) {}
+    range( Val value, Val margin ) :
+        min( value - margin ),
+        max( value + margin ) {}
+
+    Val min;
+    Val max;
```

Comment:
I'd probably define an operator that compares a value to the ranges so you can cut the number of `CHECK` statements by half.